src/worker.rs

```diff
@@ -149,6 +162,21 @@ impl Worker {
                         } else {
                             None
                         };
+                        let resp = if allowed_to_mutate.is_some()
+                            && !allowed_to_mutate.unwrap()
+                            && mutated
+                        {
+                            ValidationResponse {
+                                allowed: false,
+                                status: Some(ValidationResponseStatus {
+                                    message: Some(format!("Request rejected by policy {}. The policy attempted to mutate the request, but it is currently configured to not allow mutations.", &req.policy_id)),
+                                    code: None,
+                                }),
+                                ..resp
+                            }
+                        } else {
+                            resp
+                        };
```

Comment:
I think what follows is a bit more readable:

```suggestion
                        let resp = if mutated && allowed_to_mutate == Some(false)
                        {
                            ValidationResponse {
                                allowed: false,
                                status: Some(ValidationResponseStatus {
                                    message: Some(format!("Request rejected by policy {}. The policy attempted to mutate the request, but it is currently configured to not allow mutations.", &req.policy_id)),
                                    code: None,
                                }),
                                ..resp
                            }
                        } else {
                            resp
                        };
```

Reply:
@ereslibre , your suggestion to make this if more readable does not compile due a issue of comparing different types (&mut Option<bool> and Option<bool>). One solution is to implement the PartialEq trait for the types. But I was not sure if implementing the trait for type from std lib is a best practice or not. What do you think?

Anyway, trying to improve the readability I've extracted the code into a function.